app/src/main/java/com/github/lookupgroup27/lookup/ui/profile/Collection.kt

```diff
@@ -1,29 +1,142 @@
 
-import android.widget.Toast
-import androidx.compose.runtime.Composable
-import androidx.compose.runtime.LaunchedEffect
+import android.util.Log
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.blur
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil.compose.rememberAsyncImagePainter
+import com.github.lookupgroup27.lookup.R
 import com.github.lookupgroup27.lookup.ui.navigation.NavigationActions
-import com.github.lookupgroup27.lookup.ui.overview.SampleScreen
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.storage.FirebaseStorage
+import kotlinx.coroutines.launch
 
-/**
- * A simple placeholder Composable function for the Collection screen. This will prevent errors when
- * navigating to the "Your Collection" section. It will be implemented with real functionality
- * later.
- */
 @Composable
-fun CollectionScreen(navigationActions: NavigationActions) {
+fun CollectionScreen(
+    navigationActions: NavigationActions,
+    testImageUrls: List<String> = emptyList()
+) {
   val context = LocalContext.current
+  val scope = rememberCoroutineScope()
+  val imageUrls = remember { mutableStateListOf<String>().apply { addAll(testImageUrls) } }
 
-  // Show a Toast message when the CollectionScreen is displayed
-  LaunchedEffect(Unit) {
-    Toast.makeText(context, "Collection screen is not yet implemented", Toast.LENGTH_SHORT).show()
+  if (testImageUrls.isEmpty()) {
+    LaunchedEffect(Unit) {
+      val user = FirebaseAuth.getInstance().currentUser
+      val userEmail = user?.email ?: ""
+      if (userEmail.isNotEmpty()) {
+        val folderPath = "images/$userEmail/"
+        scope.launch {
+          val storage = FirebaseStorage.getInstance()
+          val imagesRef = storage.getReference().child(folderPath)
+
+          imagesRef.listAll().addOnSuccessListener { result ->
+            if (result.items.isEmpty()) {
+              Log.d("FirebaseImageDebug", "No items found in the '$folderPath' folder")
+            } else {
+              result.items
+                  .sortedBy { it.name }
+                  .forEach { item ->
+                    item.downloadUrl.addOnSuccessListener { uri -> imageUrls.add(uri.toString()) }
```

Comment:
Ensure that any retrieved image URLs are valid and non-null before adding them to imageUrls to avoid potential null pointer exceptions in the UI.